### ISSUE TITLE
[bug] `target with the same name already exists` error since `GLOBAL to conan targets`

### ISSUE THREAD

**Issue by climblinne** [conan-io/conan#6480]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04
  * Compiler+version: GCC8
  * Conan version: 1.22.0
  * CMake: 3.14.5
  * Python version: 3.6.8

### Steps to reproduce (Include if Applicable)
I use conan workspace to generate my workspace.

I get the following error:

```
CMake Error at /home/default/work/ws/obj/gcc-8-x86_64-Release/pf_std_os/build/conanbuildinfo.cmake:192 (add_library):
  add_library cannot create imported target "CONAN_PKG::my_modules"
  because another target with the same name already exists.
```

It seems, that https://github.com/conan-io/conan/pull/6438 (issue: https://github.com/conan-io/conan/issues/3482) make the problem.

Conan Version 1.21.2 works fine.
When changing entries for libraries, which are similar to this

```
add_library(CONAN_PKG::my_modules INTERFACE IMPORTED GLOBAL)
```
to the following
```
add_library(CONAN_PKG::my_modules INTERFACE IMPORTED)
```
everything works again.

**Comment by jgsogo** [comment]
Hi, @climblinne 

I'm trying to narrow down this issue looking for the use-cases where it is failing. I assume you don't have two packages with the same name `my_modules`, so it should be related to the combination of _workspaces_ (using `add_subdirectory`) and `cmake` generator.

Let me know if this resembles your configuration: you have two packages in your workspace consuming the same `my_modules` library, so both of them are including a `conanbuildinfo.cmake` file and running `conan_basic_setup(TARGETS)`, both calls create the `CONAN_PKG::my_modules` target, now with the `GLOBAL` keyword CMake raises the error.

If this is the problem, it is inherent to the current workspaces implementation where we let each package consume its own `conanbuildinfo.cmake` 😓 , there is a lot of work to do in the workspaces feature...

I need to try it, but a possible workaround for the workspaces use-case could be to use the `cmake_find_package` generator with `find_package(my_modules)` (I expect _find_package_ to isolate these targets despite of the GLOBAL) and avoid the `TARGETS` argument in the `conan_basic_setup()` call.

**Comment by climblinne** [comment]
Hello @jgsogo,

yes you are right. There are no libraries with two times the same name. 

We use `conan_basic_setup(TARGETS)` in our projects/libraries.  Yes, I have in my current workspace two executables and multiple libraries, with dependency to a cached library, where the problem occurs.

For me it would be very hard to change  from the current way in CMake. Would it be possible to check if  the "target" is already defined, before doing again the definition. Something like:

```
if not <variable>
   set <variable>
   set(_CONAN_PKG_LIBS_...)
   string(REPLACE " " ";" _CONAN_PKG_LIBS_...)
   conan_package_library_targets(...)
   add_library(...)
endif
```

**Comment by jgsogo** [comment]
Yes, that kind of check can be done, for sure. Maybe that could be a hotfix for a minor release while we fix the underlying workspace issue. Let me share it with the rest of the team and I'll come back soon

**Comment by memsharded** [comment]
Reverted, will be released in patch 1.22.1
